cogs/ahk/ahk.py

```diff
@@ -725,39 +818,46 @@ def check(inter: disnake.MessageInteraction):
                     "Object-Oriented": tags["Object-Oriented"],
                 },
             ),
-        )
+            SingleSelect(
+                "Do any of the following apply to your post? Skip if none apply.",
+                self.bot,
+                thread,
+                {
+                    "Written by AI": tags["ChatGPT / AI"],
+                    "Someone else wrote it": tags["Someone Else"],
+                },
+            ),
+        ]
 
         for question in questions:
-            picked = await ask(*question)
-
-            # None signifies a timeout
-            # add whatever was picked, if anything
-            if picked is None:
-                break
+            message = await question.ask(message, inter)
 
-            # add tag to list if we picked one
-            # anything else, probably means a skip
-            if isinstance(picked, disnake.ForumTag):
-                added_tags.append(picked)
+            try:
+                result = await question.interact(message)
+            except asyncio.TimeoutError:
+                if message is not None:
+                    await message.delete()
+                return
 
-        # just delete and stop if we timed out
-        if not added_tags:
-            await message.delete()
-            return
+            added_tags.update(result)
 
-        await thread.edit(applied_tags=added_tags)
+        from random import sample
 
-        content = (
-            "Thanks for tagging your post!\nYou can change the tags at any time by using `/retag`\n"
-        )
+        added_tags_list = list(added_tags)
```

Comment:
@itayVolk if you want/have interest, it's possible to come up with some kind of algorithm to prioritize some tags over others if more than 5 is chosen. Here I just naively cast to a list and take the first five, something more intelligent could be done in theory